Source/WebCore/loader/archive/cf/LegacyWebArchive.cpp

```diff
@@ -79,16 +80,44 @@ static const CFStringRef LegacyWebArchiveResourceTextEncodingNameKey = CFSTR("We
 static const CFStringRef LegacyWebArchiveResourceResponseKey = CFSTR("WebResourceResponse");
 static const CFStringRef LegacyWebArchiveResourceResponseVersionKey = CFSTR("WebResourceResponseVersion");
 
+static bool isUnreservedURICharacter(UChar character)
+{
+    return isASCIIAlphanumeric(character) || character == '-' || character == '.' || character == '_' || character == '~';
+}
+
+static String encodeURIComponentForFileName(StringView input)
```

Comment:
I think this name is confusing. Normally, `encodeURIComponent("a b c")` returns a%20b%20c, but this function does not actually encodeURIComponent, it transforms the filename to be filesystem-safe.

Reply:
The `encodeURIComponentForFileName` here means encoding a component extracted from URI for file name use, not encoding a component for URI use (as `encodeURIComponent` in JavaScript).
I see it could be a bit confusing; I am going to change it to `getFileNameFromURIComponent`.